core/jvm/src/main/scala/scalaz/zio/blocking/Blocking.scala

```diff
@@ -68,10 +68,22 @@ object Blocking extends Serializable {
           } finally lock.unlock()
 
         val interruptThread: UIO[Unit] =
-          ZIO.effectTotal(withMutex(thread.get match {
-            case None         => ()
-            case Some(thread) => thread.interrupt()
-          }))
+          ZIO.effectTotal {
+            var looping = true
+            var n       = 0L
+            val base    = 2L
+            while (looping) {
+              withMutex(thread.get match {
+                case None         => looping = false; ()
+                case Some(thread) => thread.interrupt()
+              })
+
+              if (looping) {
+                n += 1
+                Thread.sleep(base * n)
```

Comment:
Last note: probably good to bound this at something like 50ms. We don't want threads that fail to interrupt to cause this fiber to lock up a thread for long periods.